java/bulk-export/bulk-export-core/src/main/java/sleeper/bulkexport/core/recordretrieval/LeafPartitionRecordRetriever.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.bulkexport.core.recordretrieval;
+
+import sleeper.bulkexport.core.model.BulkExportLeafPartitionQuery;
+import sleeper.core.iterator.CloseableIterator;
+import sleeper.core.record.Record;
+
+/**
+ * Interface for a LeafPartitionRecordRetriever.
+ */
+@FunctionalInterface
+public interface LeafPartitionRecordRetriever {
```

Comment:
It looks like this is only used in tests. Since we'll use Rust to perform the bulk export, and we'll never return the records to the user through Java, I think we can remove this?

The same applies for BulkExportRecordRetrievalException.

Reply:
Removed.